lib/plugins/aws/package/compile/events/cloudFront/index.js

```diff
@@ -168,13 +168,18 @@ class AwsCompileCloudFrontEvents {
             // Retain Lambda@Edge functions to avoid issues when removing the CloudFormation stack
             Object.assign(Resources[lambdaFunctionLogicalId], { DeletionPolicy: 'Retain' });
 
-            const lambdaVersionLogicalId = _.findKey(Resources, {
-              Type: 'AWS::Lambda::Version',
-              Properties: {
-                FunctionName: {
-                  Ref: this.provider.naming.getLambdaLogicalId(functionName),
-                },
-              },
+            const lambdaVersionLogicalId = Object.keys(Resources).find(key => {
+              const r = Resources[key];
+              if (r.Type !== 'AWS::Lambda::Version') return false;
+              if (!r.Properties) return false;
+              if (!r.Properties.FunctionName) return false;
+              if (
+                r.Properties.FunctionName.Ref !==
+                this.provider.naming.getLambdaLogicalId(functionName)
```

Comment:
As I see, we have logical id already accessible at `lambdaFunctionLogicalId` variable